jerry-core/ecma/base/ecma-helpers-number.c

```diff
@@ -333,6 +333,22 @@ ecma_number_is_infinity (ecma_number_t num) /**< ecma-number */
           && (fraction == 0));
 } /* ecma_number_is_infinity */
 
+/**
+ * Check if number is finite
+ *
+ * @return true  - if number is finite
+ *         false - if number is NaN or infinity
+ */
+extern inline bool JERRY_ATTR_ALWAYS_INLINE
+ecma_number_is_finite (ecma_number_t num) /**< ecma-number */
+{
+#ifdef WIN32
+  return isfinite (num);
+#else /* !WIN32 */
+  return __builtin_isfinite (num);
```

Comment:
How well is this supported by compilers outside of GCC? Is it possible that some compilers don't provide this builtin?

Reply:
Of course I can add a fallback for compilers are not GCC, Clang or MSVC. But do / should we support any other comiler?

Reply:
I added the fallback.